airflow/executors/kubernetes_executor.py

```diff
@@ -497,16 +541,16 @@ def clear_not_launched_queued_tasks(self, session=None) -> None:
 
             # Try run_id first
             kwargs["label_selector"] += ",run_id=" + pod_generator.make_safe_label_value(ti.run_id)
-            pod_list = self.kube_client.list_namespaced_pod(self.kube_config.kube_namespace, **kwargs)
-            if pod_list.items:
+            pod_list = self._list_pods(kwargs)
+            if pod_list:
```

Comment:
```suggestion
            for _ in self._list_pods(kwargs):
```
if you accept above suggestion, you'd have to do this